src/dotnet/CarbonAware.Tools.WattTimeClient/WattTimeClientException.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CarbonAware.Tools.WattTimeClient
+{
+    public class WattTimeClientException : Exception
+    {
+        /// <summary>
+        /// Creates a new instance of the <see cref="WattTimeClientException"/> class.
+        /// </summary>
+        /// <param name="message">The error message supplied.</param>
+        /// <param name="response">The response object generating this exception.</param>
+        public WattTimeClientException(string message, HttpResponseMessage? response) : base(message)
+        {
+            this.Response = response;
+        }
+
+        /// <summary>
+        /// Gets the status code for the exception.  See remarks for the status codes that can be returned.
+        /// </summary>
+        /// <remarks>
+        /// 400:  Returned when the lattitude/longitude provided aren't associated with a known balancing authority.
+        /// 401:  Returned when no authorization header is passed.  You should not expect to receive this status code.
```

Comment:
>/// 401:  Returned when no authorization header is passed.  You should not expect to receive this status code. [](http://example.com/codeflow?start=8&length=110)

According to the docs, 401 is also returned on Token Expiration too -- might necessitate a 'token refresh' case being detected (in the future) 

[https://www.watttime.org/api-documentation/#login-amp-obtain-token](https://www.watttime.org/api-documentation/#login-amp-obtain-token)

Reply:
Ok, I now see the code in WattTimeClient.cs that does handle the retry